src/jvmMain/kotlin/io/beatmaps/controllers/map.kt

```diff
@@ -118,16 +119,40 @@ fun Route.mapController() {
             headerTemplate = {
                 try {
                     transaction {
-                        Beatmap.joinVersions().select {
-                            Beatmap.id eq it.key.toInt(16) and Beatmap.deletedAt.isNull()
-                        }.limit(1).complexToBeatmap().map { MapDetail.from(it, cdnPrefix()) }.firstOrNull()
+                        Beatmap
+                            .joinCollaborators()
+                            .joinVersions()
+                            .select {
+                                Beatmap.id eq it.key.toInt(16) and Beatmap.deletedAt.isNull()
+                            }.limit(1).complexToBeatmap().map { MapDetail.from(it, cdnPrefix()) }.firstOrNull()
                     }?.let {
                         meta("og:type", "website")
                         meta("og:site_name", "BeatSaver")
                         meta("og:title", it.name)
                         meta("og:url", "${Config.siteBase()}/maps/${it.id}")
                         meta("og:image", it.publishedVersion()?.coverURL)
                         meta("og:description", it.description.take(400))
+                        
+                        // Joining mappers together for the og:author field so that:
+                        // 1. Uploader will be first
+                        // 2. All non-last collaborators will be joined with ", "
+                        // 3. The last collaborator will be joined with ", and "  
```

Comment:
The last item in a list shouldn't have the comma though

Reply:
Not sure if anyone wants to play code golf with me but here's an approach using reduce to allow this to be customised
```kotlin
val authors = (listOf(it.uploader) + (it.collaborators ?: listOf())).map { u -> u.name }.toMutableList()
val authorString = authors.reduceIndexed { index, acc, s -> "$acc${if (index == authors.lastIndex) " and" else ","} $s" }
```